src/pages/search.tsx

```diff
@@ -158,17 +158,36 @@ const Search: NextPage<SearchProps> = ({ translations, chaptersData }): JSX.Elem
     [],
   );
 
+  // a ref to know weather this is the initial search request made when the user loads the page or not
+  const isInitialSearch = useRef(true);
+
   // listen to any changes in the API params and call BE on change.
   useEffect(() => {
     // only when the search query has a value we call the API.
     if (debouncedSearchQuery) {
-      getResults(debouncedSearchQuery, currentPage, selectedTranslations, selectedLanguages);
+      // we don't want to reset pagination when the user reloads the page with a ?page={number} in the url query
+      if (!isInitialSearch.current) setCurrentPage(1);
+
+      getResults(
+        debouncedSearchQuery,
+        // if it is the initial search request, use the page number in the url, otherwise, reset it
+        isInitialSearch.current ? currentPage : 1,
+        selectedTranslations,
+        selectedLanguages,
+      );
+
+      // if it was the initial request, update the ref
+      if (isInitialSearch.current) isInitialSearch.current = false;
     }
-  }, [currentPage, debouncedSearchQuery, getResults, selectedLanguages, selectedTranslations]);
+    // we don't want to run this effect when currentPage is changed
+    // because we are already handeling this in onPageChange
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [debouncedSearchQuery, getResults, selectedLanguages, selectedTranslations]);
 
-  const onPageChange = (page: number) => {
+  const onPageChange = async (page: number) => {
     logEvent('search_page_number_change');
     setCurrentPage(page);
+    await getResults(debouncedSearchQuery, page, selectedTranslations, selectedLanguages);
```

Comment:
It shows this message since `getResults` doesn't return a `Promise`.

<img width="871" alt="Screen Shot 2022-09-01 at 4 20 54 PM" src="https://user-images.githubusercontent.com/15169499/187879581-bfb4285b-cb59-4c42-9a91-cf67ebdf68b7.png">

Reply:
Done